src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalMessagesOpt.java

```diff
@@ -52,8 +51,13 @@ public void getComponentMessages(MessageCacheItem cacheItem) {
 			Map<String, String> messages = (JSONObject) jsonObject.get("messages");
 			cacheItem.setCacheItem(messages, null, System.currentTimeMillis(), null);
 		} catch (Exception e) {
-			logger.error("Failed to get offline messages for product: " + dto.getProductID() + " " + dto.getVersion() +
-					", component: " + dto.getComponent() + ", locale: " + dto.getLocale() + ", exception: " + e.getMessage());
+			String msg = "Failed to get offline messages for product: " + dto.getProductID() + " " + dto.getVersion() +
+					", component: " + dto.getComponent() + ", locale: " + dto.getLocale() + ", exception: " + e.getMessage();
+			if (!ConstantsKeys.SOURCE.equals(dto.getLocale())) {
+				logger.error(msg);
+			}else{
+				logger.debug(msg);
+			}
```

Comment:
```suggestion
			String msg = "Failed to get offline messages for product: {} {}, component: {}, locale: {}, exception: {}";
			if (!ConstantsKeys.SOURCE.equals(dto.getLocale())) {
				logger.error(msg, dto.getProductID(), dto.getVersion(), dto.getComponent(), dto.getLocale(), e.getMessage());
			}else{
				logger.debug(msg, dto.getProductID(), dto.getVersion(), dto.getComponent(), dto.getLocale(), e.getMessage());
			}
```

Reply:
I think change like this the code looks not so simple.